modules/storages/app/components/storages/admin/forms/oauth_client_form_component.html.erb

```diff
@@ -2,23 +2,81 @@
   render(Primer::Beta::Text.new(tag: :div, test_selector: 'storage-oauth-client-form')) do
     primer_form_with(
       model: oauth_client,
-      url: admin_settings_storage_oauth_client_path(storage)
+      url: admin_settings_storage_oauth_client_path(storage),
+      method: form_method,
+      data: {
+        controller: 'storages--oauth-client-form',
+        application_target: 'dynamic',
+        'storages--oauth-client-form-client-id-missing-value': t(:"storages.instructions.one_drive.missing_client_id_for_redirect_uri")
+      }
     ) do |form|
       flex_layout do |oauth_client_row|
         oauth_client_row.with_row(mb: 3) do
           render(Primer::Beta::Text.new(font_weight: :semibold)) { I18n.t("storages.file_storage_view.#{storage.short_provider_type}_oauth") }
         end
 
         oauth_client_row.with_row(mb: 3) do
-          render(Primer::Beta::Text.new(font_weight: :light)) { storage_provider_credentials_copy_instructions }
+          render(Primer::Beta::Text.new(font_weight: :light, test_selector: 'storage-provider-credentials-instructions')) { storage_provider_credentials_instructions }
         end
 
         oauth_client_row.with_row(mb: 3) do
-          render(Storages::Admin::OAuthClientForm.new(form, storage:))
+          render(
+            Storages::Admin::OAuthClientForm.new(
+              form,
+              storage:,
+              client_id_input_options: {
+                data: {
+                  'storages--oauth-client-form-target': 'clientId',
+                  action: 'input->storages--oauth-client-form#toggleSubmitButtonDisabled '\
+                          'input->storages--oauth-client-form#setRedirectUriValue'
+                }
+              },
+              client_secret_input_options: {
+                data: {
+                  'storages--oauth-client-form-target': 'clientSecret',
+                  action: 'input->storages--oauth-client-form#toggleSubmitButtonDisabled'
+                }
+              }
+            )
```

Comment:
Is there a reason you're explicitly specifying the `input` event as the listener here? Stimulus provides an "Event Shorthand" notation which lets you rely on default events for different elements (unless you explicitly want to listen on a different event (e.g. `change`):

https://stimulus.hotwired.dev/reference/actions#event-shorthand

<img width="931" alt="image" src="https://github.com/opf/openproject/assets/61627014/6f00ca00-b88c-48a6-8600-269f0bfb7ac2">

Reply:
Just out of preference as I find the explicit definitions more clear and easier to follow

Reply:
Got it, thought it was some special case.